Batfish fails unzipping a zip without directory entries

Zipping the example testrig: `zip -r -D example.zip example/`
Produces a zip without directory entries.:
```
spencer$ zipinfo example.zip 
Archive:  example.zip
Zip file size: 110323 bytes, number of entries: 21
-rw-r--r--  3.0 unx      375 tx defN 18-Apr-26 08:58 example/node_roles.json
-rw-r--r--  3.0 unx     5990 tx defN 17-Nov-06 17:01 example/example-topology.json
-rw-r--r--  3.0 unx      934 tx defN 18-Mar-27 10:06 example/testrig_positions
-rw-r--r--  3.0 unx     3467 tx defN 18-Jan-09 16:42 example/configs/as3border1.cfg
-rw-r--r--  3.0 unx     1706 tx defN 18-Jan-09 16:42 example/configs/as1core1.cfg
...
```

When initializing a testrig using that zip, `batfish` throws the following exception:
```
batfish> init-testrig example.zip
WMS:uploadTestrig exception for apikey:00000000000000000000000000000000 in container:cp_0d0b9db3-8b3c-438e-93e3-1ce7845dc24e, testrig:tr_9a900ab2-2360-4b33-9440-af39cebb9bf1; exception:org.batfish.common.BatfishException: Error unzipping to output file: '/private/var/folders/5g/5c3c39k17xz7npm4cj8bnnxm0000gp/T/tr3110976160923530596/example/node_roles.json'
	at org.batfish.common.util.UnzipUtility.extractFile(UnzipUtility.java:32)
	at org.batfish.common.util.UnzipUtility.unzip(UnzipUtility.java:86)
	at org.batfish.coordinator.WorkMgr.uploadTestrig(WorkMgr.java:1674)
	at org.batfish.coordinator.WorkMgrService.uploadTestrig(WorkMgrService.java:1799)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384)
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:222)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:565)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:545)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.FileNotFoundException: /private/var/folders/5g/5c3c39k17xz7npm4cj8bnnxm0000gp/T/tr3110976160923530596/example/node_roles.json (No such file or directory)
	at java.io.FileOutputStream.open0(Native Method)
	at java.io.FileOutputStream.open(FileOutputStream.java:270)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at org.batfish.common.util.UnzipUtility.extractFile(UnzipUtility.java:29)
	... 29 more
Error in PostData: failure Error unzipping to output file: '/private/var/folders/5g/5c3c39k17xz7npm4cj8bnnxm0000gp/T/tr3110976160923530596/example/node_roles.json'
```

--- a/projects/batfish-common-protocol/src/main/java/org/batfish/common/util/UnzipUtility.java
+++ b/projects/batfish-common-protocol/src/main/java/org/batfish/common/util/UnzipUtility.java
@@ -82,6 +82,13 @@ public final class UnzipUtility {
               throw new IOException("Unable to make directory " + outputPath);
             }
           } else {
+            // Make sure parent directories exist, in case the zip does not contain dir entries
+            File parentDir = outputPath.toFile().getParentFile();
+            if (!parentDir.exists()) {
+              if (!parentDir.mkdirs()) {
+                throw new IOException("Unable to make directory " + parentDir.getPath());
+              }
+            }
             // Extract the file.
             extractFile(zipIn, outputPath);
           }

--- a/projects/batfish-common-protocol/src/test/java/org/batfish/common/util/UnzipUtilityTest.java
+++ b/projects/batfish-common-protocol/src/test/java/org/batfish/common/util/UnzipUtilityTest.java
@@ -46,7 +46,6 @@ public class UnzipUtilityTest {
     File directoriesMissing = _folder.newFile("directories_missing");
     try (FileOutputStream fos = new FileOutputStream(directoriesMissing);
         ZipOutputStream out = new ZipOutputStream(fos)) {
-      out.putNextEntry(new ZipEntry("missing_dir/child_dir/"));
       out.putNextEntry(new ZipEntry("missing_dir/child_dir/file.txt"));
       out.write(contents);
     }

